source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -907,6 +908,30 @@ message TransformationOutlineFunction {
 
 }
 
+message TransformationPermuteFunctionParameters {
+
+  // A transformation that, given a non-entry-point function taking n
+  // parameters and a permutation of the set [1, n]:
+  //   - Introduces a new function type that is the same as the original
+  //     function's type but with the order of arguments permuted
+  //     (only if it doesn't already exist)
+  //   - Changes the type of the function to this type
+  //   - Adjusts all calls to the function so that their arguments are permuted
+
+  // Function, whose parameters will be permuted
+  uint32 function_id = 1;
+
+  // An id for a new function type that might be created
+  uint32 fresh_type_id = 2;
+
+  // OpFunctionCall instructions that call our function
+  repeated InstructionDescriptor call_site = 3;
+
+  // Permutation for the function's parameters
```

Comment:
Can you explain this a little more in the comment?  I suppose there are expected to be n elements, such that i maps to permutation[i]?  (And again I think this fits better with a [0, n-1] model.)

Reply:
Fixed in 19a2a10